wagtail/documents/urls.py

```diff
@@ -1,9 +1,11 @@
-from django.conf.urls import url
+from django.urls import path, re_path
 
 from wagtail.documents.views import serve
 
 urlpatterns = [
-    url(r'^(\d+)/(.*)$', serve.serve, name='wagtaildocs_serve'),
-    url(r'^authenticate_with_password/(\d+)/$', serve.authenticate_with_password,
+    re_path(r'^(\d+)/(.*)$', serve.serve, name='wagtaildocs_serve'),
```

Comment:
I tried to change this to a `path` style url but couldn't get it to work. AFAIK there is no easy way to do an optional path argument (for the `(.*)` part).

Reply:
Originally the document serve view just had the ID, then the filename was added as an optional cosmetic thing - but this was changed in #3528 so that passing a correct filename is now required. Logically, then, any tests that don't pass a filename should have started failing with #3528 (unless they somehow set a blank filename in the database too...?)

Probably not essential to investigate this as part of the current PR, unless you really want to :-)